spec/requests/api/v1/attendances_spec.rb

```diff
@@ -38,7 +38,11 @@
   describe 'GET /api/v1/attendances' do
     include_context 'with correct api version header'
 
-    before { sign_in logged_in_user }
+    before do
+      sign_in logged_in_user
+
+      allow(Time).to receive(:current).and_return(week_current_date)
```

Comment:
The spec fixed the current week to a fixed date, causing the attendances to be created within the ranges of the fixed date. Also, note that for the application internally use `Time.current` when there is no `filter_date` supplied.

We didn't encounter the error last week because the current week fixed date was last week's date which coincides with `Time.current` at the time.

But as we enter into this week, the value of `Time.current` has moved to this week but the current week date in the spec is still last week since it is fixed. Hence, attendances will be created using `last week's` and `two weeks ago from last week's` dates. So when we filter with `Time.current` which fall this week, we won't get any record back

So the solution is to also fix `Time.current` so it always coincides with the fixed current week date